app/src/main/java/ir/noori/learningandroid/MyApplication.kt

```diff
@@ -2,14 +2,17 @@ package ir.noori.learningandroid
 
 import android.app.Application
 import android.content.Context
+import android.os.Handler
 
 open class MyApplication : Application() {
     companion object {
         lateinit var context: Context
+        lateinit var handler: Handler
```

Comment:
Why do you define the `handler` here? Couldn't you define it in the repository class?

Reply:
Unfortunately, in our projects, we have a Handler for managing UI thread that is defined and set once in the Application class. This handler is then accessed globally throughout the project.

Reply:
I should replace the `handlers` with `Coroutine`. doesn't it?